nav2_map_server/include/nav2_map_server/map_2d/map_saver_2d.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2020 Samsung Research Russia
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__MAP_2D__MAP_SAVER_2D_HPP_
+#define NAV2_MAP_SERVER__MAP_2D__MAP_SAVER_2D_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav2_msgs/srv/save_map.hpp"
+
+#include "nav2_map_server/map_saver_core.hpp"
+#include "nav2_map_server/map_2d/map_io_2d.hpp"
+
+namespace nav2_map_server
+{
+
+/**
+ * @class nav2_map_server::MapSaver<nav_msgs::msg::OccupancyGrid>
+ * @brief A class that provides map saving methods and services
+ * for OccupancyGrid maps via. SaveMap service
+ * SaveMap service default name : save_map
+ */
+template<>
+class MapSaver<nav_msgs::msg::OccupancyGrid>: public nav2_util::LifecycleNode
```

Comment:
Why is this file templated? Its called `map_saver_2d` of which the type of a 2D map is `OccupancyGrid`. And the `template<>` is empty, so there's not even an argument to use if you wanted to. 

If anything, this `MapSaver` shouldn't be specific to 2D or 3D and the template should be that message type and the service type -- e.g. `template<typename MsgT, typename ServiceT>` where MsgT is the Pointcloud/occ grid and the ServiceT is the SaveMap/SaveMap3D. Then this file wouldn't be specifically for 2D or 3D (in naming or directory location). I'm not necessarily pushing for you to do that right now, but from looking at the header file functions, I don't see where the 2D vs 3D actually differs here.